flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/CatalogPropertiesUtil.java

```diff
@@ -472,7 +510,8 @@ private static int getCount(Map<String, String> map, String key, String suffix)
         final String escapedSeparator = Pattern.quote(SEPARATOR);
         final Pattern pattern =
                 Pattern.compile(
-                        escapedKey
+                        "^"
```

Comment:
Why change this?

Reply:
Before, for the key `generic.schema.1.name` and `schema.1.name` will all pass this pattern, then the column count will misleading the key extractor. It actually expect the `^` match here. So for the `generic.schema.1.name` it will return 0. Then, we can use the fallback key to get.